### ISSUE TITLE
upstream version and fork on gitlab

### ISSUE THREAD

**Issue by oe1rsa** [VK3FNG/soundmodem#1]
For those who are interested:

The "official" upstream version from Thomas most likely is
https://gitlab.com/tsailer/soundmodem

I based a fork on this version:
https://gitlab.com/packetradio/soundmodem
that has been moved forward to gtk3. My fork, however, comes without windows support (flex driver) and has no diagnostic interface.

( I am reading the linux-hams kernel mailinglist.)

**Comment by n2ygk** [comment]
upstream looks like it has some improvements also not captured in this copy.

**Comment by oe1rsa** [comment]
I guess by upstream you mean: https://gitlab.com/tsailer/soundmodem, yes?
What is "this" copy? Do you mean my fork? https://gitlab.com/packetradio/soundmodem
or https://github.com/VK3FNG/soundmodem.git ?
So specifically which improvements do you mean? Can you pls. be more explicit?

**Comment by n2ygk** [comment]
@oe1rsa yes by upstream I mean Thomas Sailer's on gitlab. By "this" I mean this repo, https://github.com/VK3FNG/soundmodem. I've cloned both repos and see differences. I've now looked at packetradio and see that one seems to have the most activity (thanks!).  Unfortunately, Debian is currently using https://gitlab.com/tsailer/soundmodem as its upstream with a number of patches as well: https://salsa.debian.org/debian-hamradio-team/soundmodem/-/blob/master/debian/upstream/metadata
so the variety of versions  is confusing (not to mention the one or two other packages that also call themselves soundmodem that are not based on Thomas's code at all).

However, the diag interface is a required piece of this package in order to facilitate adjusting audio levels, so my reading of your initial comment in this issue is that packetradio/soundmodemconfig will not let one use the scope feature?

I've been using soundmodem for many years under CentOS and switched to Debian and recently obtained a [DINAH](https://hamprojects.info/dinah/) C-Media based USB sound card that uses the HID interface for PTT and I was debugging why it was unable to properly use /dev/hidraw0 and found that it was missing the #define for HAVE_LINUX_HIDRAW_H:

```
soundmodem-diffs$ grep -lr HAVE_LINUX_HID .
./vk3fng/soundcard/ptt.c
./vk3fng/config.h
./vk3fng/config.h.in
./vk3fng/config.status
./vk3fng/config.log
./tsailer/soundcard/ptt.c
./packetradio/soundcard/ptt.c
soundmodem-diffs$ diff tsailer/config.h.in vk3fng/config.h.in 
126a127,129
> /* Define to 1 if you have the <linux/hidraw.h> header file. */
> #undef HAVE_LINUX_HIDRAW_H
> 
```
I determined that this was because HAVE_LINUX_HIDRAW_H was not set in the Debian build. This is something that autoconf should be set up to #define.

Here's where it's done right in this repo's (vk3fng's) version:
```diff
soundmodem-diffs$ git diff vk3fng/configure.ac packetradio/configure.ac |grep hidraw
-AC_CHECK_HEADERS(pty.h getopt.h syslog.h sched.h linux/sockios.h sys/ioccom.h linux/ppdev.h linux/hidraw.h)
```

This change was added at some point:
```
vk3fng$ git blame -L58,58 configure.ac 
^f4c4438 configure.in (Matt Grice 2018-06-17 22:19:22 +1000 58) AC_CHECK_HEADERS(pty.h getopt.h syslog.h sched.h linux/sockios.h sys/ioccom.h linux/ppdev.h linux/hidraw.h)
```
Which is the initial commit in this repo so it is unclear where it comes from. Maybe a later version of tsailer's code that was distributed pre-git via sourceforge or something.

I suppose the easy debian fix (for me) is yet another debian package patch that doesn't get propagated upstream, but the upstream they are using is unmaintained and the "right" thing is to make sure this support is in the "good" upstream!

**Comment by oe1rsa** [comment]
@n2ygk I agree, the diag interface looks like a convenient means for looking at the real signals, however it would need yet more time for implementation. I am not sure, however, if the interface is really required.

I was able to do completely without the diag gui by using a simple gnuradio companion gui for the task. This can even be accomplished while running the soundmodem daemon. (The trick is to set up the alsa device for multiple concurrent access.)

I do not believe, however, it is a good idea to make soundmodem depend on gnuradio for the sole task of level adjustment. What do you think, would it be enough to provide a documentation, or is it really worth to implement a scope for the sole purpose of this task?

Regarding the HIDRAW, I will try to look into it as soon as I have some time. Since I am using serial devices, the issue did not appear to me (yet).

**Comment by n2ygk** [comment]
@oe1rsa Documentation using gnuradio would be helpful, although I prefer the diag interface since it also decodes packets, etc.

**Comment by n2ygk** [comment]
@oe1rsa it also looks like Tom is still updating his repo: https://gitlab.com/tsailer/soundmodem/activity. But I guess you know that since you submitted a PR there: https://gitlab.com/tsailer/soundmodem/-/merge_requests/1

**Comment by n2ygk** [comment]
@oe1rsa See https://gitlab.com/tsailer/soundmodem/-/merge_requests/3
